### ISSUE TITLE
Question about the aug_entropy_loss

### ISSUE THREAD

**Issue by zsxm1998** [MikeWangWZHL/PAPO#20]
Thanks for your excellent work!

I have a question about the computation of the `aug_entropy_loss`, i.e. the part of the Double Entropy Regularization in the paper. The `aug_entropy_loss` is computed from `model_inputs["aug_log_probs"]`, which is computed under the `torch.no_grad()` function `compute_log_prob`. That means no gradient will backward to the model, so maybe this loss is useless. And that's maybe why only using single entropy loss on the corrupted policy sequence does not prevent the collapse.

May I ask for your perspective on this matter?

Thanks.

**Comment by MikeWangWZHL** [comment]
Thanks for the great question! Yes your observation is accurate, in current implementation the "aug_entropy_loss" does not directly affect the gradients; the major regularization should come from the entropy on pi_theta; empirically we did find that adding this aug_entropy_loss slightly improves performance which we are not entirely sure on the reason (may be due to some numerical stability); 
In order to make the aug_entropy_loss affect the gradients I believe we need to do an additional forward on the masked sequence during the actor updating steps; we haven't try this setting but would be interesting to see results! Thanks!
